include/WCSimWCDigitizer.hh

```diff
@@ -38,7 +38,10 @@ public:
   void SetDigitizerPEPrecision      (double precision) { DigitizerPEPrecision     = precision; }; ///< Override the default digitizer charge resolution (p.e.)
 
   double Truncate(double value, double precision) {
-    if(precision < 1E-10) return value;
+      if(precision < 1E-10) return value;
+      float ratio=value / precision;
+      // Result of cast is undefined if value is outside range of int. Just do nothing in that case.
+      if(ratio>2147483647 || ratio<-2147483648) return value;
```

Comment:
This looks good, but can you combine it with 20c056940603647a0bda8e9f37fbc62fef97ef4c from @ast0815? i.e. use `long int`, and the relevant max allowed values, instead?

Reply:
In fact I'll go further! I'll add this:

 ```
if(ratio>std::numeric_limits<long long>::max() || ratio<std::numeric_limits<long long>::min() ) return value;
    return precision * (long long)(value / precision)
```